pkg/k8s/resource.go

```diff
@@ -0,0 +1,62 @@
+package k8s
```

Comment:
WDYT about putting this under a `resource` package and remove all the appearances of the "Resource" word in the functions below? So they'd be called like `resource.Render()`, etc.

Reply:
Hmmm, I like that way but still we can't remove the `Resource` word from the fn's right? How would they named be instead?

Reply:
```go
foo := resource.New()
foo = resource.NewNamespaced()
foo.Render()
```

Reply:
Done 👍

Reply:
To [avoid stuttering](https://blog.golang.org/package-names) I would get rid of all mentioning of "resource" in that file, besides the package name itself. So instead of `type KubernetesResource` I would put `type Kubernetes` and instead of `func (r KubernetesResource) RenderResource()` it'd be `func (r Kubernetes) Render()`. You can even use an import alias to make things extra clear: `k8sResource "github.com/linkerd/linkerd2/pkg/k8s/resource"`

Reply:
Good Suggestion! _(Also great blog suggestion, made things clear)_

> import alias to make things extra clear: k8sResource "github.com/linkerd/linkerd2/pkg/k8s/resource"
Does not seem that useful, so skipped this!